### PR TITLE
mupdf: 1.17.0 -> 1.18.0, pymupdf: 1.17.6 -> 1.18.0, llpp 32 -> 33

### LINKED ISSUE #90910: Vulnerability roundup 85: mupdf-1.17.0: 1 advisory [7.5]
[search](https://search.nix.gsc.io/?q=mupdf&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=mupdf+in%3Apath&type=Code)

* [ ] [CVE-2017-5991](https://nvd.nist.gov/vuln/detail/CVE-2017-5991) CVSSv3=7.5 (nixos-unstable)

Scanned versions: nixos-unstable: 22c98819ccd. May contain false positives.

Cc @fpletz
Cc @vrthra

### LINKED ISSUE #100316: Vulnerability roundup 94: mupdf-1.17.0: 1 advisory [5.5]
[search](https://search.nix.gsc.io/?q=mupdf&i=fosho&repos=NixOS-nixpkgs), [files](https://github.com/NixOS/nixpkgs/search?utf8=%E2%9C%93&q=mupdf+in%3Apath&type=Code)

* [ ] [CVE-2020-26519](https://nvd.nist.gov/vuln/detail/CVE-2020-26519) CVSSv3=5.5 (nixos-20.09, nixos-unstable)

Scanned versions: nixos-20.09: 0b8799ecaaf; nixos-unstable: 420f89ceb26.

Cc @fpletz
Cc @vrthra

### PR DESCRIPTION
**Pull request by wamserma**
also drops the patch for shared libs (not needed any more)

fixes https://github.com/NixOS/nixpkgs/issues/90910
fixes https://github.com/NixOS/nixpkgs/issues/100316 (unstable)

###### Motivation for this change

Upstream release and security issues mentioned above.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "NIXPKGS_ALLOW_INSECURE=1 nixpkgs-review pr 100441"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wamserma** [comment]
this bundles a number of package updates similar to https://github.com/NixOS/nixpkgs/pull/88919 to avoid a broken state on master

**Comment by wamserma** [comment]
@ofborg eval
@ofborg build mupdf mupdf-1-17 pymupdf llpp k2pdfopt jfbview

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -22488,6 +22488,7 @@ in
   pdfsam-basic = callPackage ../applications/misc/pdfsam-basic { };
 
   mupdf = callPackage ../applications/misc/mupdf { };
+  mupdf-1-17 = callPackage ../applications/misc/mupdf/mupdf-1-17.nix { };
```
Comment (jonringer) [inline pkgs/top-level/all-packages.nix:22491]: generally, pinned versions use an underscore `_`
```suggestion
  mupdf_1_17 = callPackage ../applications/misc/mupdf/1.17.nix { };
```

Comment (wamserma) [reply pkgs/top-level/all-packages.nix]: PR updated.

**Comment by wamserma** [comment]
@jonringer While you're at it: Can you add the security label to the PR?

File: pkgs/applications/misc/mupdf/default.nix [hunk 2]
```diff
@@ -13,27 +14,43 @@ let
...
+  version = "1.18.0";
   pname = "mupdf";
 
   src = fetchurl {
     url = "https://mupdf.com/downloads/archive/${pname}-${version}-source.tar.gz";
-    sha256 = "13nl9nrcx2awz9l83mlv2psi1lmn3hdnfwxvwgwiwbxlkjl3zqq0";
+    sha256 = "0rljl44y8p8hgaqializlyrgpij1wbnrzyp0ll5kcg7w05nylq48";
   };
 
   patches =
```
Comment (andir) [inline pkgs/applications/misc/mupdf/default.nix:25]: Can you comment why you picked these patches? The upstream repo shows a few more patches that were added after the release. It isn't clear (from this discussion & the commit message) why these were picked.

File: pkgs/applications/misc/k2pdfopt/default.nix [hunk 3]
```diff
@@ -2,7 +2,7 @@
 , cmake, pkgconfig, zlib, libpng, makeWrapper
 , enableGSL ? true, gsl
 , enableGhostScript ? true, ghostscript
-, enableMuPDF ? true, mupdf-1-17
+, enableMuPDF ? true, mupdf_1_17
```
Comment (andir) [inline pkgs/applications/misc/k2pdfopt/default.nix:5]: Can you squash that fixup into cf9e181cbdf7bd9c015d7c32eb936a9ed315e3f4 ?

File: pkgs/applications/misc/mupdf/default.nix (hunk 2 above)
Comment (wamserma) [reply pkgs/applications/misc/mupdf/default.nix:25]: These seemed to be sensible post-release bugfixes that I deemed worth including. At the time of creating this PR, there were only a few patches.
We should not block this PR on the inclusion of further patches or dropping the patches already included, as there is already a RC for 1.18.1 which I prefer to put into a new PR.

Comment (andir) [reply pkgs/applications/misc/mupdf/default.nix:25]: Ok, sounds reasonable enough.

**Comment by andir** [comment]
This looks good as is. Could you also add the relevant CVE identifiers (or links to our tracking issues) to the relevant commits?

**Comment by wamserma** [comment]
> This looks good as is. Could you also add the relevant CVE identifiers (or links to our tracking issues) to the relevant commits?

I'll put the CVEs, as e.g. https://github.com/NixOS/nixpkgs/issues/100316 can only be closed once this is also backported.

**Review by jonringer — APPROVED** [review]
diff LGTM

```
https://github.com/NixOS/nixpkgs/pull/100441
1 package marked as broken and skipped:
mupdf_1_17

10 packages built:
cups-filters llpp mupdf python27Packages.pymupdf python37Packages.pymupdf python38Packages.pymupdf splix system-config-printer termpdfpy zathura
```

**Comment by timokau** [comment]
This (commit e13120bb098f25ac552570c2f9ea5908ef102a50 in particular) broke the `zathura` pdf viewer at runtime:

```
$ zathura some-file.pdf
error: Could not load plugin '/nix/store/l73f0x9xwcngvmq0wngdk77if0zd3n99-zathura-with-plugins-0.4.5/lib/zathura/libpdf-mupdf.so' (/nix/store/l73f0x9xwcngvmq0wngdk77if0zd3n99-zathura-with-plugins-0.4.5/lib/zathura/libpdf-mupdf.so: undefined symbol: jpeg_resync_to_restart).
error: Could not determine file type.
```

zathura then displays a black window without any of the pdf's contents.

**Comment by timokau** [comment]
Also CC @globin, the maintainer of zathura.

**Pull request merged**